src/utils/xhr-utils.js

```diff
@@ -414,11 +414,10 @@ export function assertSuccess(response) {
  * Returns a promise resolving to a string identity token if the element
  * contains the 'crossorigin' attribute and the amp-viewer-assistance extension
  * is present. Resolves to undefined otherwise.
- * @param {!Window} win
  * @param {!Element} element
  * @return {!Promise<string|undefined>}
  */
-export function getViewerAuthTokenIfAvailable(win, element) {
+export function getViewerAuthTokenIfAvailable(element) {
```

Comment:
An alternative fix is to resolve with empty string instead of `undefined`, and then change the check in `batchFetchJsonFor` to compare against undefined instead of truthy. I.e.

```js
return Services.viewerAssistanceForDocOrNull(element).then(va => {
  userAssert(va, 'crossorigin="amp-viewer-auth-token-post" requires amp-viewer-assistance extension.');
  return va.getIdTokenPromise();
}).then(token => token || '').catch(() => ''); // Return empty string if token is not available.
```

This preserves the current semantics by keeping the "crossorigin" attribute encapsulated in this helper function. WDYT?

Reply:
I am okay with having the token return an empty string instead of undefined without a token, but it's important to still transform the request without a token based on the presence of the `crossorigin` attribute -- not the identity token.

If I'm understanding your proposal correctly, the token would only be undefined in cases where `getViewerAuthTokenIfAvailable` isn't called at all. However, `getViewerAuthTokenIfAvailable` will always be called in `amp-state` and `amp-list` before being passed to the batched-json call, so it will always transform the request into a POST because the `opt_token !== undefined`.

Reply:
Here's a clearer snippet of what I'm suggesting:
```js
if (crossOriginAttr 
    && crossOriginAttr.trim() === 'amp-viewer-auth-token-via-post') {
  return Services.viewerAssistanceForDocOrNull(element).then(va => {
    userAssert(va, 'crossorigin="amp-viewer-auth-token-post" '
        + 'requires amp-viewer-assistance extension.');
    return va.getIdTokenPromise();
  })
  // If crossorigin attribute is present, resolve with token or empty string.
  .then(token => token || '').catch(() => ''); 
}
// If crossorigin attribute is missing, always resolve with undefined.
return Promise.resolve(undefined);
```
Would that work?

Reply:
Ah yes, that makes sense. This solution should work! Thanks.

Reply:
Just made the changes and added some tests. Thanks for the suggestion!